pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -101,97 +111,139 @@ func (e *eksRefreshController) refreshClusterUpstreamSpec(cluster *mgmtv3.Cluste
 	// If upstreamSpec is nil then the syncing failed for some reason. This is reported to the user, and this function returns at the end of this if-statement.
 	// If upstreamSpec is non-nil then the syncing occurred as expected, but the node groups have health issues that are reported to the user.
 	// In this second case, the message is set on the Updated condition, but execution continues because the sync was successful.
-	upstreamSpec, err := GetComparableUpstreamSpec(e.secretsCache, cluster)
+	upstreamConfig, err := getComparableUpstreamSpec(c.secretsCache, cluster)
 	if err != nil {
 		var statusErr error
 		var syncFailed string
-		if upstreamSpec == nil {
+		if upstreamConfig == nil {
 			syncFailed = ": syncing failed"
 		}
 		cluster = cluster.DeepCopy()
 		apimgmtv3.ClusterConditionUpdated.False(cluster)
 		apimgmtv3.ClusterConditionUpdated.Message(cluster, fmt.Sprintf("[Syncing error%s] %s", syncFailed, err.Error()))
 
-		cluster, statusErr = e.clusterClient.Update(cluster)
+		cluster, statusErr = c.clusterClient.Update(cluster)
 		if statusErr != nil {
 			return cluster, statusErr
 		}
-
-		if upstreamSpec == nil {
-			return cluster, err
-		}
 	} else if strings.Contains(apimgmtv3.ClusterConditionUpdated.GetMessage(cluster), "[Syncing error") {
 		cluster = cluster.DeepCopy()
 		apimgmtv3.ClusterConditionUpdated.True(cluster)
 		apimgmtv3.ClusterConditionUpdated.Message(cluster, "")
-		cluster, err = e.clusterClient.Update(cluster)
+		cluster, err = c.clusterClient.Update(cluster)
 		if err != nil {
 			return cluster, err
 		}
 	}
+	if upstreamConfig == nil {
+		return cluster, err
+	}
 
-	if !reflect.DeepEqual(cluster.Status.EKSStatus.UpstreamSpec, upstreamSpec) {
+	var clusterCloudConfig, upstreamCloudConfig, appliedCloudConfig, statusCloudConfig interface{}
+	switch {
+	case cluster.Spec.AKSConfig != nil:
+		clusterCloudConfig = cluster.Spec.AKSConfig
+		appliedCloudConfig = cluster.Status.AppliedSpec.AKSConfig
+		statusCloudConfig = cluster.Status.AKSStatus.UpstreamSpec
+		upstreamCloudConfig = upstreamConfig.aksConfig
+	case cluster.Spec.EKSConfig != nil:
+		clusterCloudConfig = cluster.Spec.EKSConfig
+		appliedCloudConfig = cluster.Status.AppliedSpec.EKSConfig
+		statusCloudConfig = cluster.Status.EKSStatus.UpstreamSpec
+		upstreamCloudConfig = upstreamConfig.eksConfig
+	}
+
+	if !reflect.DeepEqual(statusCloudConfig, upstreamCloudConfig) {
 		logrus.Infof("updating cluster [%s], upstream change detected", cluster.Name)
 		cluster = cluster.DeepCopy()
-		cluster.Status.EKSStatus.UpstreamSpec = upstreamSpec
```

Comment:
Why has this been removed? Now the cluster is being updated but nothing has changed on the cluster. The pointer is just referencing a different struct now.

Reply:
Ahh, you are right, I think that I fixed that now, look on L#158